testsuite/features/support/env.rb

```diff
@@ -127,31 +127,72 @@ def capybara_register_driver
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
 end
 
-# embed a screenshot after each failed scenario
+# Embed a screenshot after each failed scenario
 After do |scenario|
   current_epoch = Time.new.to_i
   log "This scenario took: #{current_epoch - @scenario_start_time} seconds"
   if scenario.failed?
     begin
-      Dir.mkdir('screenshots') unless File.directory?('screenshots')
-      path = "screenshots/#{scenario.name.tr(' ./', '_')}.png"
-      # only click on Details when we have errors during bootstrapping and more Details available
-      click_button('Details') if has_content?('Bootstrap Minions') && has_content?('Details')
-      # a Timeout::Error may be raised while a page is still (re)loading
-      find('#page-body', wait: 3) if scenario.exception.is_a?(Timeout::Error)
+      if web_session_is_active?
+        handle_screenshot_and_relog(scenario, current_epoch)
+      else
+        warn 'Page is not visible; unable to take a screenshot.'
+      end
+    ensure
+      print_server_logs
+    end
+  end
+  page.instance_variable_set(:@touched, false)
+end
+
+# Test is web session is open
+def web_session_is_active?
+  # When no Web Session is open, current_url is equal to data:,
+  return false if current_url.empty? || current_url == 'data:,'
+  page.has_selector?('header', wait: 0) || page.has_selector?('#username-field', wait: 0)
```

Comment:
I removed the wait time for this test. It could be risky but will speedup a lot the failing scenario without web session open. For me, if we reach this point, it means our scenario already failed on a timeout if we have a web session active.

Former code : 
```ruby
def web_session_is_active?
  page.has_selector?('header', wait: Capybara.default_max_wait_time) ||
    page.has_selector?('#username-field', wait: Capybara.default_max_wait_time)
end
```

Reply:
I could add back the wait option because the current url check should speed up the previously describe issue.